### ISSUE TITLE
Request for safe mkfifo function

### ISSUE THREAD

**Issue by sector-f** [nix-rust/nix#602]
As far as I know, `mkfifo(3)` is just a wrapper around `mknod(3)`. Regardless, a FIFO can be created with `mknod`. 

I think a wrapper function (placed in `stat.rs`) would look something like this:

````rust
pub fn mkfifo<P: ?Sized + NixPath>(path: &P, perm: Mode) -> Result<()> {
    let res = try!(path.with_nix_path(|cstr| {
        unsafe {
            ffi::mknod(cstr.as_ptr(), nix::sys::stat::S_IFIFO.bits() as mode_t | perm.bits() as mode_t, 0)
        }
    }));

    Errno::result(res).map(drop)
}
````

Alternatively, you could also wrap `nix`'s existing safe `mknod` function like I did in this code:

````rust
use nix;
use nix::sys::stat::{mknod, Mode, S_IFIFO};
use nix::NixPath;

pub fn mkfifo<P: ?Sized + NixPath>(path: &P, perm: Mode) -> nix::Result<()> {
    mknod(path, S_IFIFO, perm, 0)
}
````

**Comment by asomers** [comment]
Only on Linux is `mkfifo` a wrapper around `mknod`.  Your proposed implementation would be unnecessarily unportable.  What's wrong with using the regular `mkfifo` function?  That one is standardized.

**Comment by sector-f** [comment]
> Only on Linux is mkfifo a wrapper around mknod.

Come to think of it, that does sound familiar. My mistake. 

> What's wrong with using the regular mkfifo function? That one is standardized.

Probably nothing; I'm just being forgetful and/or ignorant.

**Comment by Susurrus** [comment]
@sector-f Would you be willing to contribute a PR adding a safe wrapper for mkfifo? `libc` already has the FFI definition for it, so all it takes is adding a function for it here in `nix`. I'd be happy to mentor you through it.

**Comment by Susurrus** [comment]
Closed by #774
